plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,160 @@
+    local FileManager = require("apps/filemanager/filemanager")
+    FileManager:onClose()
+    if not file_path then
+        self.context = ""
+        self.file_path = ""
+    else
+        self.file_path = file_path
+        self:readFile(file_path)
+    end
+    self.input = InputDialog:new{
+        title =  _("Basic text editor"),
+        input = self.context,
+        text_height = Screen:getHeight() * 0.5,
+        width = self.width or Screen:getWidth(),
+        input_type = "string",
+        enter_callback = function ()
+            return '\n'
+        end,
+        buttons = {{
+        {
+            text = _("Open"),
+            callback = function()
+                self:chooseFile()
+            end,
+        },
+        {
+            text = _("Save"),
+            callback = function()
+                self:saveFile(self.file_path)
+            end,
+        },{
+            text = _("Quit"),
+            callback = function()
+                UIManager:show(ConfirmBox:new{
+                    text = _("Are you sure that you want to quit editor? All unsaved changes will be lost."),
+                    ok_text = _("Quit"),
+                    ok_callback = function()
+                        self.context = ""
+                        self.file_path = ""
+                        UIManager:close(self.input)
+                    end
+                })
+
+            end,
+        },
+        {
+            text = _("Pg_up"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollUp()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        },
+        {
+            text = _("Pg_dwn"),
+            callback = function()
+                self.input._input_widget.text_widget:scrollDown()
+                UIManager:setDirty(self.input, function()
+                    return "ui", self.input.dimen
+                end)
+            end,
+        }
+
+    }},
+    }
+    UIManager:show(self.input)
+    self.input:onShowKeyboard()
+end
+
+
+function TextEditor:chooseFile()
+    self.input:onClose()
+    local path_chooser = PathChooser:new{
+        title = _("Choose file. Long press to confirm"),
+        height = Screen:getHeight(),
+        path = util.realpath(DataStorage:getDataDir()),
+        show_hidden = G_reader_settings:readSetting("show_hidden"),
+        file_filter = function() return true end,
+        onConfirm = function(file_path)
+            logger.dbg("TextEditor: selected file_file_path " .. file_path )
+            self.file_path = file_path
+            self:readFile(self.file_path)
+            self.input:onShowKeyboard()
+        end
+    }
+    UIManager:show(path_chooser)
+end
+
+function TextEditor:readFile(file_path)
+    logger.dbg("TextEditor: reading file: " .. file_path)
+    local file = assert(io.open(file_path, "rb"))
```

Comment:
Please don't use asserts just like that unless it's dbg.assert

Reply:
ok